vm/state_migration/src/lib.rs

```diff
@@ -0,0 +1,259 @@
+                self.migrations.len(),
+            ));
+        }
+
+        let cpus = num_cpus::get();
+        let chan_size = cpus / 2;
+
+        log::info!(
+            "Using {} CPUs for migration and channel size of {}",
+            cpus,
+            chan_size
+        );
+
+        let pool = rayon::ThreadPoolBuilder::new()
+            .thread_name(|id| format!("nv12 migration thread: {}", id))
+            .num_threads(cpus)
+            .build()
+            .map_err(MigrationError::ThreadPoolCreation)?;
+
+        let (state_tx, state_rx) = crossbeam_channel::bounded(chan_size);
+        let (job_tx, job_rx) = crossbeam_channel::bounded(chan_size);
+
+        pool.scope(|s| {
+            let store_clone = store.clone();
+
+            s.spawn(move |_| {
+                actors_in
+                    .for_each(|addr, state| {
+                        state_tx
+                            .send((addr, state.clone()))
+                            .expect("failed sending actor state through channel");
+                        Ok(())
+                    })
+                    .expect("Failed iterating over actor state");
+            });
+
+            s.spawn(move |scope| {
+                while let Ok((address, state)) = state_rx.recv() {
+                    let job_tx = job_tx.clone();
+                    let store_clone = store_clone.clone();
+                    let migrator = self.migrations.get(&state.code).cloned().unwrap();
+                    scope.spawn(move |_| {
+                        let job = MigrationJob {
+                            address,
+                            actor_state: state,
+                            actor_migration: migrator,
+                        };
+
+                        let job_output = job.run(store_clone, prior_epoch).unwrap_or_else(|e| {
+                            panic!(
+                                "failed executing job for address: {}, Reason: {}",
+                                address, e
+                            )
+                        });
+
+                        job_tx.send(job_output).unwrap_or_else(|_| {
+                            panic!("failed sending job output for address: {}", address)
+                        });
+                    });
+                }
+                drop(job_tx);
+            });
+
+            while let Ok(job_output) = job_rx.recv() {
+                let MigrationJobOutput {
+                    address,
+                    actor_state,
+                } = job_output;
+                actors_out
+                    .set_actor(&address, actor_state)
+                    .unwrap_or_else(|e| {
+                        panic!(
+                            "Failed setting new actor state at given address: {}, Reason: {}",
+                            address, e
+                        )
+                    });
+            }
+        });
+
+        let root_cid = actors_out
```

Comment:
You can remove the assignment and the `;` to just return

Reply:
I wanted to keep it explicit on what is being returned for readability.